appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -866,12 +924,25 @@ Blockly.LexicalVariable.makeLegalIdentifier = function(ident) {
 //    (1) add prefix argument,
 //    (2) handle local declaration statements/expressions, and
 //    (3) treat prefixes correctly when they're used.
+// TODO: I think this breaks the single responsibility principle. Consider
+//   splitting it into two functions, since this is only called internally.
+// TODO: Might be good to split the recursive part from the non-recursive part.
```

Comment:
I don't think that this is possible, or at least, that it would be any more sane than what's already here.

Reply:
> that it would be any more sane than what's already here.
Truth. Removed.